In [ ]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

In [17]:
import pandas as pd
df=pd.read_csv("Amazon-Product-Reviews - Amazon Product Review (1).csv")

In [18]:
df.head()

marketplace  customer_id       review_id  product_id  product_parent  \
0          US     11555559  R1QXC7AHHJBQ3O  B00IKPX4GY         2693241   
1          US     31469372  R175VSRV6ZETOP  B00IKPYKWG         2693241   
2          US     26843895  R2HRFF78MWGY19  B00IKPW0UA         2693241   
3          US     19844868   R8Q39WPKYVSTX  B00LCHSHMS         2693241   
4          US      1189852  R3RL4C8YP2ZCJL  B00IKPZ5V6         2693241   

                           product_title product_category  star_rating  \
0  Fire HD 7, 7" HD Display, Wi-Fi, 8 GB               PC            5   
1  Fire HD 7, 7" HD Display, Wi-Fi, 8 GB               PC            3   
2  Fire HD 7, 7" HD Display, Wi-Fi, 8 GB               PC            5   
3  Fire HD 7, 7" HD Display, Wi-Fi, 8 GB               PC            4   
4  Fire HD 7, 7" HD Display, Wi-Fi, 8 GB               PC            5   

   helpful_votes  total_votes vine verified_purchase  \
0              0            0    N                 Y   
1              0            0    N                 N   
2              0            0    N                 Y   
3              0            0    N                 N   
4              0            0    N                 Y   

                                     review_headline  \
0                                         Five Stars   
1  Lots of ads Slow processing speed Occasionally...   
2                            Well thought out device   
3  Not all apps/games we were looking forward to ...   
4                                         Five Stars   

                                         review_body review_date  sentiment  
0                                      Great love it  2015-08-31          1  
1  Lots of ads<br />Slow processing speed<br />Oc...  2015-08-31          0  
2  Excellent unit.  The versatility of this table...  2015-08-31          1  
3  I bought this on Amazon Prime so I ended up bu...  2015-08-31          1  
4  All Amazon products continue to meet my expect...  2015-08-31          1

In [19]:
# Import necessary libraries

# sklearn.model_selection is used for splitting data into training and testing sets
from sklearn.model_selection import train_test_split

# sklearn.feature_extraction.text provides tools for converting text data into numerical feature vectors (TF-IDF here)
from sklearn.feature_extraction.text import TfidfVectorizer

# sklearn.metrics contains functions to evaluate model performance (accuracy_score, classification_report)
from sklearn.metrics import accuracy_score, classification_report

# sklearn.ensemble includes ensemble learning methods such as RandomForestClassifier
from sklearn.ensemble import RandomForestClassifier


In [20]:
#Prepare the data 
X=df['review_body']
y=df['sentiment']

In [21]:
#Split the data into training and testing sets
X_train,X_test, y_train, y_test=train_test_split(X,y, test_size=0.2, random_state=42)

In [22]:
# Example: Fill NaN values with an empty string
# X_train_filled = X_train.fillna('')
# X_test_filled = X_test.fillna('')

In [23]:
df['review_body'].fillna('', inplace=True)  # Replace NaN with empty string


In [24]:
from sklearn.feature_extraction.text import TfidfVectorizer
# this code snippet demonstrates how to use TfidfVectorizer to preprocess and convert text data into a numerical format suitable for machine learning models, particularly for tasks like text classification or sentiment analysis. \
# The resulting TF-IDF vectors (X_train_vect) can be used as input features for training machine learning models.

# Vectorize the text data using TF-IDF
vectorizer = TfidfVectorizer(max_features=1000)
X_train_vect = vectorizer.fit_transform(df['review_body'])  # Assuming 'review_body' is the column name containing text data in your DataFrame




In [25]:
X_test_vect=vectorizer.transform(X_test)

In [26]:
# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X_train_vect, df['sentiment'], test_size=0.2, random_state=42)


In [27]:
#Training the classifier
from sklearn.svm import SVC
classifier=SVC(kernel='linear', random_state=42)
classifier.fit(X_train, y_train)

SVC(kernel='linear', random_state=42)

In [28]:
#Making predictions 
y_pred=classifier.predict(X_test)

In [29]:
#Evaluating the model
accuracy=accuracy_score(y_test, y_pred)
report=classification_report(y_test, y_pred)

In [16]:
#Print the evaluation matrics
print(f"Accuracy:{accuracy}")
print(f"Classification Report:n\{report}")

Accuracy:0.9087520259319287
Classification Report:n\              precision    recall  f1-score   support

           0       0.77      0.64      0.70      1018
           1       0.93      0.96      0.95      5152

    accuracy                           0.91      6170
   macro avg       0.85      0.80      0.82      6170
weighted avg       0.90      0.91      0.91      6170



# Conclusion

The model performs well overall, particularly in identifying positive sentiment (`1`), with high precision and recall.
However, there is some room for improvement in identifying negative sentiment (`0`), where the precision and recall are relatively lower compared to the positive class. 
Overall, the model's performance is strong, achieving an accuracy of approximately 90.88% on the test dataset.